lab_3_keywords_textrank/main.py

```diff
@@ -173,7 +203,14 @@ def extract_pairs(tokens: tuple[int, ...], window_length: int) -> Optional[tuple
     In case of corrupt input data, None is returned:
     tokens must not be empty, window lengths must be integer, window lengths cannot be less than 2.
     """
-    pass
+    if not(isinstance(window_length, int) and window_length >= 2 and tokens):
+        return None
+    pairs = set()
+    for idx1, token1 in enumerate(tokens):
+        for idx2, token2 in enumerate(tokens):
```

Comment:
it is good, but you could make it perfect by iterating with `token2` over not all of the tokens but only those tokens that have indices greater than `idx1` and smaller thatn `idx1 + window_length`
it would be more efficient

Reply:
yeah, it's a really good point, thank you! done

Reply:
also, i've just accidentally pushed the report file again, sorry